infra/envs/prod/main.tf

```diff
@@ -0,0 +1,55 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+locals {
+  # Prefix is used in the staging environement for workspaces, don't touch me here.
+  prefix = "prod"
+  # Profile that will be used to select which account to deploy to.
+  profile = "prod"
+  # Choose the region where this infrastructure should be deployed.
+  region = "us-east-1"
+  # Add environment specific tags
+  tags = merge(module.common.tags, {
+    environment = "prod"
+    description = "Application resources created in production environment"
+    
+  })
+
+}
+
+terraform {
+
+  required_version = ">=1.2.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~>4.20.1"
+    }
+  }
+
+  # Terraform does not allow interpolation here, values must be hardcoded.
+ 
+  # backend "s3" {
+  #   bucket         = "ACCOUNT_ID-REGION-tf-state"
+  #   key            = "terraform/prod/terraform.tfstate"
+  #   region         = "REGION"
+  #   encrypt        = "true"
+  #   dynamodb_table = "tf_state_locks"
+  # }
+
+}
+
+provider "aws" {
+  region  = local.region
+  profile = local.profile
+  default_tags {
+    tags = local.tags
+  }
+}
+
+module "common" {
+  source = "../../common"
+}
```

Comment:
We do. Since we are using directories to create strong separation between environments this a root module which has a separate state from the other environment's as well as the boostrap/account/ it mirrors. We could dry it up using workspaces as our environment separation method, although not recommended, by using terragrunt, a DIY solution or option 4?

Reply:
Got it. Thanks for explaining. Let's keep what you've set up here.